prepare-host: login on github when using curl

ffromani opened the pull request:
To avoid github API throttling, let traviskube consume
the github account env vars, if provided.

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
prepare-host: login on github when using curl

To avoid github API throttling, let traviskube consume
the github account env vars, if provided.

Signed-off-by: Francesco Romani <fromani@redhat.com>

MarSik commented on ci/prepare-host:
```diff
@@ -5,6 +5,14 @@ set -e
+_curl() {
+	local auth_opts=""
+	if [ -n "${GITHUB_USER}" ] && [ -n "${GITHUB_TOKEN}" ]; then
+		auth_opts="-u ${GITHUB_USER}:${GITHUB_TOKEN}"
```
You do not need user iirc. You can use the simpler -H "Authorization: token TOKEN" I think. I even used ?authorization_token= query argument in one of my projects.

ffromani committed:
_curl: use token header, not user/pass pair

It is simpler and requires one less env var.

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
prepare-host: simplify the shell expansion

We use a full if/then branch to avoiding quoting headaches with
shell expansion.

Signed-off-by: Francesco Romani <fromani@redhat.com>

fromanirh commented:
We see from travis the auth token is passed. The failure is in the minikube lane - which we don't use, and can be fixed/investigated later.
